sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Conditions/ComparisonCondition.cs

```diff
@@ -37,19 +35,44 @@ public ComparisonCondition(string field, QueryComparisonOperator comparisonOpera
 
         public override string Stringify()
         {
+            // TODO -- generic type on this class to handle different inputs
             var conditionString = new StringBuilder();
             conditionString.Append($"{Field} {QueryConstants.ComparisonOperators[Operator]} ");
 
             // check to see if the input is numeric value -- if not, we need single quotes around Value
-            bool isNumeric = int.TryParse(Value, out int numericValue);
+            bool isNumeric;
+            bool quoteSurrounded = (Value[0] == '\'' && Value[Value.Length - 1] == '\'');
 
-            if (!isNumeric)
+            // check to see if the user has entered single quotes around the Value
+            if (quoteSurrounded)
+            {
+                isNumeric = int.TryParse(Value.Substring(1, Value.Length - 2), out _);
```

Comment:
Can you elaborate on this log?

`Value.Substring(1, Value.Length - 2)`

Stuff like this could use a code comment, btw.